### PR TITLE
[ENG-5966] 2.0.1 BE: Update permission to support edit as a WRITE contributor

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Clean-up some permissions concepts and improve testing.

## Changes

- Breaks out author assertions into separate file and organizes by permissions 
- Breaks out institutional affiliation updates and organizes by permissions 
- Simplify `can_edit` to simply a write permission

## QA Notes

N/A

## Documentation

N/A

## Side Effects

N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-5966

### CODE REVIEW

File: api_tests/preprints/views/test_preprint_institutions_relationship.py [hunk 1]
```diff
@@ -17,23 +13,37 @@ def user(self):
...
+    def write_user_with_institutional_affiliation(self, institution, preprint):
+        user = AuthUserFactory()
+        preprint.add_contributor(user, permissions=WRITE)
+        user.add_or_update_affiliated_institution(institution)
+        return user
+
+    @pytest.fixture()
+    def read_user_with_institutional_affiliation(self, institution, preprint):
         user = AuthUserFactory()
-        preprint.add_permission(user, 'admin')
```
Comment (Johnetordoff) [inline api_tests/preprints/views/test_preprint_institutions_relationship.py:22]: This lead to an odd bug in our permissions.

File: api_tests/preprints/views/test_preprint_institutions_relationship.py [hunk 2]
```diff
@@ -17,23 +13,37 @@ def user(self):
         return AuthUserFactory()
 
     @pytest.fixture()
-    def admin_with_institutional_affilation(self, institution, preprint):
+    def admin_with_institutional_affiliation(self, institution, preprint):
+        user = AuthUserFactory()
+        preprint.add_contributor(user, permissions=ADMIN)
+        user.add_or_update_affiliated_institution(institution)
+        return user
```
Comment (cslzchen) [inline api_tests/preprints/views/test_preprint_institutions_relationship.py:20]: For all the fixtures, if not too much effort, can we add more than one affiliation to the user to test more variations. e.g.

* Admin has A, B, C
* Write has B, C, D
* Read has C, E, F

**Review by cslzchen — CHANGES_REQUESTED** [review]
Looks good overall and love the unit tests 👍 . In addition to my comments,

- [x] First, how about changing `test_author_assertions_preprint_detail.py` to `test_preprint_detail_author_assertions_.py`?
- [ ] Have you check the withdrawal logic? I think we are fine since that piece is probably separate piece from the detail serializer.

File: api_tests/preprints/views/test_author_assertions_preprint_detail.py [hunk 3]
```diff
@@ -30,7 +30,15 @@ def user(self):
...
     @pytest.fixture()
     def preprint(self, user):
-        return PreprintFactory(creator=user)
+        """
+        Creator is not admin permission
+        """
+        preprint = PreprintFactory(creator=user)
+        admin = AuthUserFactory()
+        preprint.add_contributor(admin, ADMIN)
+        preprint.add_contributor(user, READ)
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply api_tests/preprints/views/test_author_assertions_preprint_detail.py:39]: It's to check that being the creator alone shouldn't be enough to change the author assertions, only any admin. I thought it was good to include. I vaguely remember you telling me to test for this, but I can't remember when or why.

**Review by cslzchen — APPROVED** [review]

File: api_tests/preprints/views/test_author_assertions_preprint_detail.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cslzchen) [reply api_tests/preprints/views/test_author_assertions_preprint_detail.py]: Ah, I vaguely remember this too and yeah, make sense.

**Pull request merged**